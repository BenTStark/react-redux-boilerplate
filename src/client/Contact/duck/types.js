const SUBMIT_MESSAGE = "SUBMIT_MESSAGE";

export default {
  SUBMIT_MESSAGE
};
