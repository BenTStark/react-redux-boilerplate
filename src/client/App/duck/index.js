import authReducer from './reducers';
export default authReducer;
export { AppOperations } from './operations';
export { default as appTypes } from './types';
