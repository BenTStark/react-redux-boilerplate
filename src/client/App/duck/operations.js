// optional for API requests
import ActionCreators from "./actions";
import AuthService from "./auth.service";
import types from "./types";

const loginRequest = () => ActionCreators.loginRequest();
const loginSuccess = authInformation => ActionCreators.loginSuccess(authInformation);
const loginError = () => ActionCreators.loginError();
const logout = () => ActionCreators.logout();

const checkLogin = () => {
  return new Promise((resolve, reject) => {
    const payload = {},
      authInformation = {};
    authInformation.result = "";
    if (AuthService.loggedIn()) {
      payload.profile = AuthService.getProfile();
      payload.accessToken = AuthService.getAccessToken();
      authInformation.result = types.LOGIN_SUCCESS;
    }
    authInformation.payload = payload;
    resolve(authInformation);
  });
};

const authentication = () => {
  // Add callback for lock's `authenticated` event
  return new Promise((resolve, reject) => {
    const payload = {},
      authInformation = {};
    authInformation.result = "";
    authInformation.payload = payload;
    AuthService.lock.on("authenticated", authResult => {
      AuthService.lock.getUserInfo(authResult.accessToken, (error, profile) => {
        if (error) {
          authInformation.result = types.LOGIN_ERROR;
          reject(authInformation);
        } else {
          AuthService.setToken(authResult.idToken, authResult.accessToken); // static method
          AuthService.setProfile(profile); // static method
          payload.profile = profile;
          payload.accessToken = authResult.accessToken;
          authInformation.result = types.LOGIN_SUCCESS;
          authInformation.payload = payload;
          AuthService.lock.hide();
          resolve(authInformation);
        }
      });
    });
    // Add callback for lock's `authorization_error` event
    AuthService.lock.on("authorization_error", error => {
      authInformation.result = types.LOGIN_ERROR;
      reject(authInformation);
    });
  });
};

const handleLogin = () => {
  AuthService.login();
};

const handleLogout = () => {
  AuthService.logout();
};

export const AppOperations = {
  handleLogin,
  loginRequest,
  loginSuccess,
  loginError,
  handleLogout,
  logout,
  checkLogin,
  authentication
};

//export default AppOperations;
