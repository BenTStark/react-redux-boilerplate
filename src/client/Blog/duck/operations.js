// optional for API requests
import ActionCreators from "./actions";
import _ from "lodash";
import moment from "moment";

const getDataUrl = "https://mirrorpi.ddns.net/cms/blogs";
const updateDataUrl = articleId =>
  `https://mirrorpi.ddns.net/cms/blogs/${articleId}`;

const getBlogArticles = articles => ActionCreators.getBlogArticles(articles);
const createBlogArticle = (articleId, article) => {
  return ActionCreators.createBlogArticle(articleId, article);
};
const deleteBlogArticle = articleId =>
  ActionCreators.deleteBlogArticle(articleId);
const setCurrentArticle = article => ActionCreators.setCurrentArticle(article);
const clearCurrentArticle = () => ActionCreators.clearCurrentArticle();
const editBlogArticle = articleId => ActionCreators.editBlogArticle(articleId);
const finishEditBlogArticle = articleId =>
  ActionCreators.finishEditBlogArticle(articleId);
const updateBlogArticle = (articleId, article) =>
  ActionCreators.updateBlogArticle(articleId, article);

const buildEditorContent = article => {
  return [
    ["title:", article.title + ","].join(" "),
    ["author:", article.author].join(" "),
    "~~~",
    article.text
  ].join("\r\n");
};
const buildArticleObject = plainText => {
  // Search for these regex in plainText to find the respective values
  const keys = [
    { key: "author", regex: /author/ },
    { key: "title", regex: /title/ }
  ];
  // target object should have keys for the article object in BlogComponent
  const result = {};
  const arr = plainText.split("~~~");
  arr[0].split(",").forEach(element => {
    keys.forEach(key => {
      if (key.regex.test(element.split(":")[0])) {
        result[key.key] = element.split(":")[1].trim();
      }
    });
  });
  // Remove \r \n
  result.text = arr[1].substring(1);
  result.isNew = false;
  return result;
};

const maxId = array => {
  return array.length ? Math.max(...array) : 1;
};

const sortBlogArticle = (a, b, articles) => {
  if (
    _.gt(
      moment.utc(articles[a].createdAt).unix(),
      moment.utc(articles[b].createdAt).unix()
    )
  ) {
    return -1;
  }
  if (
    _.lt(
      moment.utc(articles[a].createdAt).unix(),
      moment.utc(articles[b].createdAt).unix()
    )
  ) {
    return 1;
  }
  return 0;
};

export const BlogOperations = {
  getBlogArticles,
  getDataUrl,
  updateDataUrl,
  createBlogArticle,
  deleteBlogArticle,
  setCurrentArticle,
  clearCurrentArticle,
  editBlogArticle,
  finishEditBlogArticle,
  updateBlogArticle,
  buildEditorContent,
  buildArticleObject,
  maxId,
  sortBlogArticle
};
