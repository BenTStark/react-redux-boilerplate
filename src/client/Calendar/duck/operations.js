// optional for API requests
import ActionCreators from "./actions";


// This is an example how a simple operation could look like:
const defaultOperation = value => ActionCreators.defaultAction(value);





export const CalendarOperations = {
  defaultOperation
};
