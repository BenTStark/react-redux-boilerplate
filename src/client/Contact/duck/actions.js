import types from "./types";

// Actions should not do any change to the value they pass to the store.
// Any value treatment should happen before in operations.js!
// This is an example how an action should look like:
const submitMessage = contactFormResult => {
  return {
    type: types.SUBMIT_MESSAGE,
    contactFormResult: contactFormResult
  };
};

export default {
  submitMessage
};
