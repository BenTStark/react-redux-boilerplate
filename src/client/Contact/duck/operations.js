// optional for API requests
import ActionCreators from "./actions";

// This is an example how a simple operation could look like:
const submitMessage = contactFormResult =>
  ActionCreators.submitMessage(contactFormResult);

export const ContactOperations = {
  submitMessage
};
