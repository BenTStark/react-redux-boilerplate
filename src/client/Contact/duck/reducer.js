import types from "./types";

const initialState = {
  byId: [],
  byHash: {}
};

// This is an example how a simple Reducer function could look like:
function submitMessage(state, contactFormResult) {
  return {
    byId: [...state.byId, contactFormResult.id],
    byHash: {
      ...state.byHash,
      [contactFormResult.id]: contactFormResult
    }
  };
}

const contactReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.SUBMIT_MESSAGE:
      return submitMessage(state, action.contactFormResult);
    default:
      return state;
  }
};

export default contactReducer;
