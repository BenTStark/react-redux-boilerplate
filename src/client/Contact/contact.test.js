import ContactComponent from "./contact.component";
// To Do: bei App Template wird das nicht genutzt (weils aber auch nur ein Component ist)
import ContactContainer from "./contact.container";
import { ContactOperations } from "./duck/operations";


import React from "react";
import { shallow, mount } from "enzyme";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";
import MockAdapter from "axios-mock-adapter";


// Snapshot for Contact React Component
describe(">>> ContactComponent - Snapshot", () => {
  it("+++capturing Snapshot of ContactComponent", () => {
    // To Do: bei App Template wird das nicht genutzt wird hier shallow genutzt!
    const renderedValue = mount(
      <ContactComponent  />
    );
    expect(toJson(renderedValue)).toMatchSnapshot();
  });
});

// // Component Test
// describe(">>> ContactComponent --- Shallow Render React Components", () => {
//   const wrapper = shallow(<ContactComponent auth={{}} home={{}} />);
//
//   it("+++ render the COMPONENT", () => {
//     expect(wrapper.length).toEqual(1);
//   });
// });
