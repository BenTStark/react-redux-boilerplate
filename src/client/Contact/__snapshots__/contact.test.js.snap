// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> ContactComponent - Snapshot +++capturing Snapshot of ContactComponent 1`] = `
<ContactComponent>
  <div>
    <ContactFormComponent
      name="BenTStark"
      onSubmit={[Function]}
    >
      <div>
        <div>
          hallo 
          BenTStark
          . hier sind die Props: 
          {
    "name": "BenTStark"
}
        </div>
        <ReactFinalForm
          onSubmit={[Function]}
          render={[Function]}
        >
          <form
            onSubmit={[Function]}
          >
            <div>
              <label
                htmlFor="firstName"
              >
                Vorname
              </label>
              <_class
                component="input"
                name="firstName"
                type="text"
              >
                <Field
                  component="input"
                  format={[Function]}
                  name="firstName"
                  parse={[Function]}
                  reactFinalForm={
                    Object {
                      "batch": [Function],
                      "blur": [Function],
                      "change": [Function],
                      "focus": [Function],
                      "getFieldState": [Function],
                      "getRegisteredFields": [Function],
                      "getState": [Function],
                      "initialize": [Function],
                      "isValidationPaused": [Function],
                      "mutators": Object {},
                      "pauseValidation": [Function],
                      "registerField": [Function],
                      "reset": [Function],
                      "resumeValidation": [Function],
                      "setConfig": [Function],
                      "submit": [Function],
                      "subscribe": [Function],
                    }
                  }
                  type="text"
                >
                  <input
                    name="firstName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                </Field>
              </_class>
            </div>
            <div>
              <label
                htmlFor="lastName"
              >
                Nachname
              </label>
              <_class
                component="input"
                name="lastName"
                type="text"
              >
                <Field
                  component="input"
                  format={[Function]}
                  name="lastName"
                  parse={[Function]}
                  reactFinalForm={
                    Object {
                      "batch": [Function],
                      "blur": [Function],
                      "change": [Function],
                      "focus": [Function],
                      "getFieldState": [Function],
                      "getRegisteredFields": [Function],
                      "getState": [Function],
                      "initialize": [Function],
                      "isValidationPaused": [Function],
                      "mutators": Object {},
                      "pauseValidation": [Function],
                      "registerField": [Function],
                      "reset": [Function],
                      "resumeValidation": [Function],
                      "setConfig": [Function],
                      "submit": [Function],
                      "subscribe": [Function],
                    }
                  }
                  type="text"
                >
                  <input
                    name="lastName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                </Field>
              </_class>
            </div>
            <div>
              <label
                htmlFor="email"
              >
                Email
              </label>
              <_class
                component="input"
                name="email"
                type="email"
              >
                <Field
                  component="input"
                  format={[Function]}
                  name="email"
                  parse={[Function]}
                  reactFinalForm={
                    Object {
                      "batch": [Function],
                      "blur": [Function],
                      "change": [Function],
                      "focus": [Function],
                      "getFieldState": [Function],
                      "getRegisteredFields": [Function],
                      "getState": [Function],
                      "initialize": [Function],
                      "isValidationPaused": [Function],
                      "mutators": Object {},
                      "pauseValidation": [Function],
                      "registerField": [Function],
                      "reset": [Function],
                      "resumeValidation": [Function],
                      "setConfig": [Function],
                      "submit": [Function],
                      "subscribe": [Function],
                    }
                  }
                  type="email"
                >
                  <input
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="email"
                    value=""
                  />
                </Field>
              </_class>
            </div>
            <div>
              <label
                htmlFor="message"
              >
                Nachricht
              </label>
              <_class
                component="input"
                name="message"
                type="text"
              >
                <Field
                  component="input"
                  format={[Function]}
                  name="message"
                  parse={[Function]}
                  reactFinalForm={
                    Object {
                      "batch": [Function],
                      "blur": [Function],
                      "change": [Function],
                      "focus": [Function],
                      "getFieldState": [Function],
                      "getRegisteredFields": [Function],
                      "getState": [Function],
                      "initialize": [Function],
                      "isValidationPaused": [Function],
                      "mutators": Object {},
                      "pauseValidation": [Function],
                      "registerField": [Function],
                      "reset": [Function],
                      "resumeValidation": [Function],
                      "setConfig": [Function],
                      "submit": [Function],
                      "subscribe": [Function],
                    }
                  }
                  type="text"
                >
                  <input
                    name="message"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                </Field>
              </_class>
            </div>
            <button
              disabled={true}
              type="submit"
            >
              Submit
            </button>
            <button
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Clear Values
            </button>
          </form>
        </ReactFinalForm>
      </div>
    </ContactFormComponent>
  </div>
</ContactComponent>
`;
